rotkehlchen/chain/manager.py

```diff
@@ -621,10 +621,28 @@ def query_balances(
             self.query_polkadot_balances(ignore_cache=ignore_cache)
         if should_query_avax:
             self.query_avax_balances(ignore_cache=ignore_cache)
+        if ignore_cache is True:
+            self.derive_new_addresses_from_xpubs(
+                should_derive_bch_xpubs=should_query_bch,
+                should_derive_btc_xpubs=should_query_btc,
+            )
 
         self.totals = self.balances.recalculate_totals()
         return self.get_balances_update()
 
+    def derive_new_addresses_from_xpubs(
+            self,
+            should_derive_btc_xpubs: bool = False,
+            should_derive_bch_xpubs: bool = False,
+    ) -> None:
+        """Derive new addresses from xpubs.
+        """
```

Comment:
Can be changed in a different PR
```suggestion
        """Derive new addresses from xpubs."""
```

Reply:
+ 1 on this